conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureHandler.java

```diff
@@ -64,27 +67,59 @@
 
     private final RoutingHandler routingHandler;
 
-    public ConjureHandler(HttpHandler fallback) {
+    private ConjureHandler(HttpHandler fallback, List<Endpoint> endpoints) {
         this.routingHandler = Handlers.routing().setFallbackHandler(fallback);
-    }
-
-    public ConjureHandler() {
-        this(ResponseCodeHandler.HANDLE_404);
+        endpoints.forEach(this::register);
     }
 
     @Override
     public void handleRequest(HttpServerExchange exchange) throws Exception {
         routingHandler.handleRequest(exchange);
     }
 
-    @Override
-    public ConjureHandler add(Endpoint endpoint, HttpHandler handler) {
-        HttpHandler current = handler;
+    private void register(Endpoint endpoint) {
+        HttpHandler current = endpoint.handler();
         for (BiFunction<Endpoint, HttpHandler, HttpHandler> wrapper : WRAPPERS) {
             current = wrapper.apply(endpoint, current);
         }
         routingHandler.add(endpoint.method(), endpoint.template(), current);
-        return this;
     }
 
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
+
+        private final List<Endpoint> endpoints = Lists.newArrayList();
+        private HttpHandler fallback = ResponseCodeHandler.HANDLE_404;
+
+        private Builder() { }
+
+        public Builder endpoints(Endpoint value) {
+            endpoints.add(Preconditions.checkNotNull(value, "Value is required"));
+            return this;
+        }
+
+        public Builder addAllEndpoints(Iterable<Endpoint> values) {
```

Comment:
I used conjure bean style names. using `Builder endpoint(Endpoint value)` above (rather than `endpoints`) and calling this method `endpoints` without clearing existing values may be cleaner, but wouldn't match what we do for generated objects.

Reply:
I understand the arguments but I would still prefer `endpoint(Endpoint value)` and `endpoint(Endpoint... endpoints)`

Reply:
is that a worthwhile performance optimization? how much of an overhead does a 1-element array incur?